README.md

```diff
@@ -1,2 +1,307 @@
-# buildingimages
+# Building Container Images in IBM Cloud
+
 content creation around image building for containers
+
+# Prerequisites
+
+Before you get started you'll need an IBM Cloud account and need to
+install
+the
+[IBM Cloud Development Tools](https://github.com/IBM-Cloud/ibm-cloud-developer-tools).
+
+You should also clone
+the [guestbook repository](https://github.com/IBM/guestbook) as all
+image builds will be based on that.
+
+Once those are installed, you'll be able to complete the commands
+below.
+
+# Basics of Image Building
+
+Kubernetes uses standard container images to deploy applications.
+
+From the application's perspective, a container image is a single
+filesystem that is created before the application is started.
+
+From a creation perspective, container images are layers. These layers
+are built up using a Docker file. Lets look at the example guestbook
+application Docker file and pull it apart.
+
+```
+FROM golang as builder
+RUN go get github.com/codegangsta/negroni
+RUN go get github.com/gorilla/mux github.com/xyproto/simpleredis
+COPY main.go .
+RUN go build main.go
+
+FROM busybox:ubuntu-14.04
+
+COPY --from=builder /go//main /app/guestbook
+
+ADD public/index.html /app/public/index.html
+ADD public/script.js /app/public/script.js
+ADD public/style.css /app/public/style.css
+
+WORKDIR /app
+CMD ["./guestbook"]
+EXPOSE 3000
+```
+
+Docker files specific a set of commands, at each command we build a
+new layer.
+
+FROM - pulls a layer from a remote repository. The default
+repository is the docker image registry. This could be a complete OS
+image, or some partial image that used to do something small. In this
+example we pull in 2 different base images. One is a named layer for
```

Comment:
I think we need to make it clear that we're not just pulling two images for fun, its because we're doing 2 (almost) independent builds within one Dockerfile.  And the "so that it can be referenced later" isn't clear.... call out the "as" usage.

Reply:
sure.